base_optional_quick_create/model.py

```diff
@@ -50,14 +48,18 @@ def _register_hook(self, cr, ids=None):
                     model_obj.check_quick_create = True
         return True
 
+    def _register_hook(self, cr):
+        self._patch_quick_create(cr, self.search(cr, SUPERUSER_ID, []))
+        return super(IrModel, self)._register_hook(cr)
+
     def create(self, cr, uid, vals, context=None):
-        res_id = super(ir_model, self).create(cr, uid, vals, context=context)
-        self._register_hook(cr, [res_id])
+        res_id = super(IrModel, self).create(cr, uid, vals, context=context)
+        self._patch_quick_create(cr, [res_id])
```

Comment:
Does it make any sense to patch the creation method after the record has been created?